kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -512,28 +512,44 @@ protected void handleInactive(KafkaHeaderAndRequest kafkaHeaderAndRequest,
             final AtomicInteger pendingNamespacesCount = new AtomicInteger(allowedNamespaces.size());
-
+            Runnable completeOne = () -> {
+                if (pendingNamespacesCount.decrementAndGet() == 0) {
+                    topicMapFuture.complete(topicMap);
+                }
+            };
             for (String namespace : allowedNamespaces) {
-                pulsarService.getNamespaceService().getListOfPersistentTopics(NamespaceName.get(namespace))
-                        .whenComplete((topics, e) -> {
-                            if (e != null) {
-                                log.error("Failed to getListOfPersistentTopic of {}", namespace, e);
-                                topicMapFuture.completeExceptionally(e);
-                                return;
-                            }
-                            if (topicMapFuture.isCompletedExceptionally()) {
+                authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.NAMESPACE, namespace))
+                        .whenComplete((isAuthorized, ex) -> {
+                            if (ex != null) {
+                                log.error("Describe namespace authorize failed, namespace - {}. {}",
+                                        namespace, ex.getMessage());
+                                completeOne.run();
                                 return;
                             }
-                            for (String topic : topics) {
-                                final TopicName topicName = TopicName.get(topic);
-                                final String key = topicName.getPartitionedTopicName();
-                                topicMap.computeIfAbsent(
-                                        KopTopic.removeDefaultNamespacePrefix(key, namespacePrefix),
-                                        ignored -> Collections.synchronizedList(new ArrayList<>())
-                                ).add(topicName);
-                            }
-                            if (pendingNamespacesCount.decrementAndGet() == 0) {
-                                topicMapFuture.complete(topicMap);
+                            if (isAuthorized) {
+                                pulsarService.getNamespaceService()
+                                        .getListOfPersistentTopics(NamespaceName.get(namespace))
+                                        .whenComplete((topics, e) -> {
+                                            if (e != null) {
+                                                log.error("Failed to getListOfPersistentTopic of {}", namespace, e);
+                                                topicMapFuture.completeExceptionally(e);
+                                                return;
+                                            }
+                                            if (topicMapFuture.isCompletedExceptionally()) {
+                                                return;
+                                            }
```

Comment:
Could `topicMapFuture.isCompletedExceptionally()` happen here? This judgment seems unnecessary.

Reply:
I think it's used to skip the following steps when `topicMapFuture` completed exceptionally in the callback of another future.

Reply:
It's great, i get it. Thanks for your reminder. :)